### PR TITLE
adds graphql schema for program dashboard

### PR DESCRIPTION
**Pull request by hlminh2000**
**Description of changes**

- rewrites `app.js` to `app.ts` with some minor refactoring to support the change.
- adds config for `ELASTICSEARCH_HOST`
- preliminary resolver setup (will be subject to change, this is just the placeholder setup)
- adds graphql schema design for the program dashboard. New queries are:
  - `programDonorSummaryVersion`
  - `programDonorSummaryEntries`
  - `programDonorSummaryStats`

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [ ] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: src/schemas/ProgramDonorSummary/gqlTypeDefs.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import { gql } from 'apollo-server-express';
```
Comment (hlminh2000) [inline src/schemas/ProgramDonorSummary/gqlTypeDefs.ts]: this file is the main thing of this PR

File: src/schemas/ProgramDonorSummary/gqlTypeDefs.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    NO_RELEASE
+    UNKNOWN
+  }
+
+  enum DonorProcessingStatus {
+    COMPLETED
+    PROCESSING
+    REGISTERED
+    UNKNOWN
+  }
```
Comment (hlminh2000) [inline src/schemas/ProgramDonorSummary/gqlTypeDefs.ts]: @jaserud can you provide the actual values we're expecting for these Enums? :) it'd be nice if it matches up with the actual system

Comment (jaserud) [reply src/schemas/ProgramDonorSummary/gqlTypeDefs.ts]: This looks fine to me. I think there is a bit of uncertainty about how these states will be updated. I know `REGISTERED` will come from clinical but I can't say for the rest.

Comment (hlminh2000) [reply src/schemas/ProgramDonorSummary/gqlTypeDefs.ts]: how about the values for `DonorReleaseStatus` above? :)

Comment (jaserud) [reply src/schemas/ProgramDonorSummary/gqlTypeDefs.ts]: Ha, I know even less about those :stuck_out_tongue:

Comment (hlminh2000) [reply src/schemas/ProgramDonorSummary/gqlTypeDefs.ts]: right right, we don't have them defined :P

File: src/services/elasticsearch/index.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Client } from '@elastic/elasticsearch';
+import { ELASTICSEARCH_HOST } from 'config';
+
+export const createClient = () => {
```
Comment (jaserud) [inline src/services/elasticsearch/index.ts]: I'm guessing vault integration will be in future pr??

File: src/schemas/ProgramDonorSummary/resolvers.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  }
+> = (source, args, context): DonorSummaryEntry[] => {
+  const { programId } = args;
+
+  console.log('args: ', args.filters);
+
+  return [];
+};
+
+const programDonorSummaryVersionResolver: GraphQLFieldResolver<
```
Comment (jaserud) [inline src/schemas/ProgramDonorSummary/resolvers.ts]: Not sure I understand where this will come from in es?

Comment (hlminh2000) [reply src/schemas/ProgramDonorSummary/resolvers.ts]: after a discussion, we decided this isn't necessary yet :) will spec out later as need arises

**Pull request merged**